graylog2-server/src/main/java/org/graylog2/alerts/AlertImpl.java

```diff
@@ -73,34 +96,75 @@ static Builder builder() {
     public static AlertImpl create(@JsonProperty(FIELD_ID) @ObjectId @Id String id,
                                    @JsonProperty(FIELD_STREAM_ID) String streamId,
                                    @JsonProperty(FIELD_CONDITION_ID) String conditionId,
-                                   @JsonProperty(FIELD_TRIGGERED_AT) Date triggeredAt,
+                                   @JsonProperty(FIELD_TRIGGERED_AT) Date triggeredAtDate,
+                                   @JsonProperty(FIELD_RESOLVED_AT) Date resolvedAtDate,
                                    @JsonProperty(FIELD_DESCRIPTION) String description,
-                                   @JsonProperty(FIELD_CONDITION_PARAMETERS) Map<String, Object> conditionParameters) {
+                                   @JsonProperty(FIELD_CONDITION_PARAMETERS) Map<String, Object> conditionParameters,
+                                   @JsonProperty(FIELD_IS_INTERVAL) boolean isInterval) {
+        final DateTime triggeredAt = new DateTime(triggeredAtDate);
+        final DateTime resolvedAt = resolvedAtDate == null ? null : new DateTime(resolvedAtDate);
```

Comment:
The argument will be `null` when deserialising old alerts from MongoDB. Avoiding the condition, sets `resolvedAt` to `now` to all alerts that haven't got the field. I'll add the `@Nullable` annotation to the parameter and a comment to make it clear.